File: jetpack_smart_display/app/src/main/java/com/smsoft/smartdisplay/data/database/entity/Sensor.kt
package com.smsoft.smartdisplay.data.database.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.smsoft.smartdisplay.data.SensorType

@Entity(tableName = "sensors", indices = [Index("type")])
data class Sensor(
    @PrimaryKey(autoGenerate = true)
    val id: Long,
    val title: String,
    val titleIcon: String,
    val topic1: String,
    val topic1Unit: String,
    val topic1Icon: String,
    val topic2: String,
    val topic2Unit: String,
    val topic2Icon: String,
    val topic3: String,
    val topic3Unit: String,
    val topic3Icon: String,
    val topic4: String,
    val topic4Unit: String,
    val topic4Icon: String,
    @ColumnInfo(name = "type", defaultValue = "MQTT")
    val type: String
)

val emptySensor = Sensor(
    id = 0,
    title = "",
    titleIcon = "",
    topic1 = "",
    topic1Unit = "",
    topic1Icon = "",
    topic2 = "",
    topic2Unit = "",
    topic2Icon = "",
    topic3 = "",
    topic3Unit = "",
    topic3Icon = "",
    topic4 = "",
    topic4Unit = "",
    topic4Icon = "",
    type = SensorType.getDefaultId()
)

File: flutter_smart_display/lib/models/sensor.dart
import 'package:flutter_smart_display/models/sensor_type.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'sensor.freezed.dart';
part 'sensor.g.dart';

@freezed
class Sensor with _$Sensor {
  const factory Sensor({
    @JsonKey(includeIfNull: false) int? id,
    required String title,
    @Default("") String titleIcon,
    required String topic1,
    @Default("") String topic1Unit,
    @Default("") String topic1Icon,
    @Default("") String topic2,
    @Default("") String topic2Unit,
    @Default("") String topic2Icon,
    @Default("") String topic3,
    @Default("") String topic3Unit,
    @Default("") String topic3Icon,
    @Default("") String topic4,
    @Default("") String topic4Unit,
    @Default("") String topic4Icon,
    @Default("MQTT") String type,
  }) = _Sensor;

  factory Sensor.fromJson(Map<String, Object?> json) => _$SensorFromJson(json);

  static Sensor emptySensor = Sensor(
      id: null,
      title: "",
      titleIcon: "",
      topic1: "",
      topic1Unit: "",
      topic1Icon: "",
      topic2: "",
      topic2Unit: "",
      topic2Icon: "",
      topic3: "",
      topic3Unit: "",
      topic3Icon: "",
      topic4: "",
      topic4Unit: "",
      topic4Icon: "",
      type: SensorType.defaultType.name);
}
